src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -246,7 +246,8 @@ private static bool CanHaveSemicolon(SyntaxNode currentNode)
                 return true;
             }
 
-            if (currentNode is RecordDeclarationSyntax { OpenBraceToken: { IsMissing: true } })
+            if (currentNode is RecordDeclarationSyntax { OpenBraceToken: { IsMissing: true } } ||
+                currentNode is RecordStructDeclarationSyntax { OpenBraceToken: { IsMissing: true } })
```

Comment:
@jcouv i don't understand this.  why do we have two syntaxes here.  We should just have RecordDeclaration that has two kinds.  If we need an actual RecordStructDeclarationSyntax, then let's make RecordDeclarationSyntax abstract and the base type of RecordClassDeclarationSyntax and RecordStructDeclarationSyntax.  RecordDeclarationSyntax will just be so confusing in code as people won't think it means "record class".

Reply:
(note: teh most important thing here fo rme is that we not have RecordStructDeclarationSyntax.  I don't see what purpose it serves given it will be the exact same shape as RecordDeclarationSyntax.